pom.xml

```diff
@@ -74,6 +75,8 @@
     <sat.version>0.7.0</sat.version>
     <slf4j.version>1.7.21</slf4j.version>
     <xtext.version>2.17.0</xtext.version>
+    <spotless.version>1.24.3</spotless.version>
+    <spotless.check.skip>true</spotless.check.skip> <!-- Spotless disabled for now -->
```

Comment:
You can also override it on the command line; 
`mvn spotless:check -Dspotless.check.skip=false`

The idea is that the build should fail if there are formatting errors. 

Spotless binds default to the maven `verify` phase. Personally I've found it annoying to realize that my code has a formatting error _after_ compiling & running all the tests. So I have bound it to the very first phase; `initialize` . On modules with a lot of source code this overhead up front can be irritating though.

If you do not skip checking (as it is now configured) the plugin will start failing the build immediately. So I suggest we 


1. incorporate the PR
1. do a reformat of the various modules, either all or one by one with corresponding commits
1. after all code has been reformatted correctly, remove the `<spotless.check.skip>true</spotless.check.skip>` line from the pom

I think which phase it should be bound to should be discussed.

If you want a real overview of the formatting errors in the whole repo, ensure you have no local changes and then run `mvn spotless:apply`. Inspect with `git status` & `git diff`.

The config above has been copied to all relevant repo's. I am not familiar enough with the current setup to determine if it could be added to the superpom instead.

Reply:
Thank you very much for your explanation.

I second your proposed strategy. For #2 we should do it bundle by bundle to minimize the conflicts in existing PRs.

I already did a reformat of the while repository and was impressed by the number of changes.

Regarding the superpom you may at best ask @pfink for help. I am afraid I cannot help you very much with that too.